src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -0,0 +1,712 @@
+use uom::si::{
+    area::square_meter,
+    f64::*,
+    force::newton,
+    length::meter,
+    ratio::ratio,
+    velocity::meter_per_second,
+    volume::{cubic_meter, gallon},
+    volume_rate::gallon_per_second,
+};
+
+use super::rigid_body::LinearActuatedRigidBodyOnHingeAxis;
```

Comment:
Had a lot of thoughts and hesitations on this one. The rigid body implemented is already a really particular one that is hinge constrained and can be moved by a linear actuator connected to its control arm.

Generally speaking, we would have a standard non constrained rigid body accepting a force applied anywhere or a torque applied anywhere. Then a constrained rigid body would be the same but with a different update. Then a linear actuated one would be the constrained one but having a control arm to connect a linear actuator.

BUT... for now I just can't think of a different one we'd have to simulate actually as all moving parts on that plane are actually sumed up in a linear actuated rigid body.

So I'd keep it simple like this until proved otherwise by an actuator/body couple I didn't think of.  Ideas?  Can only think of the flap slat mechanism, but in all my researches I ended up not simulating the rigid body at all for flaps and going for a different approach.

Reply:
As discussed on Discord, it might be that parts of both modules are so (logically) interconnected that they belong in the same module.